genvm/metrics.go

```diff
@@ -0,0 +1,141 @@
+package vm
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/spacemeshos/go-spacemesh/metrics"
+)
+
+const namespace = "vm"
+
+var (
+	transactionDuration = metrics.NewHistogramWithBuckets(
+		"transaction_duration",
+		namespace,
+		"Duration in ns without writing updated state on disk",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000, 2, 10),
+	).WithLabelValues()
+	transactionDurationParse = metrics.NewHistogramWithBuckets(
+		"transaction_duration_parse",
+		namespace,
+		"Duration in ns for parsing a transaction",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000, 2, 10),
+	).WithLabelValues()
+	transactionDurationVerify = metrics.NewHistogramWithBuckets(
+		"transaction_duration_verify",
+		namespace,
+		"Duration in ns for verifying a transaction",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000, 2, 10),
+	).WithLabelValues()
+	transactionDurationExecute = metrics.NewHistogramWithBuckets(
+		"transaction_duration_execute",
+		namespace,
+		"Duration in ns for executing a transaction",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000, 2, 10),
+	).WithLabelValues()
+)
+
+var (
+	blockDuration = metrics.NewHistogramWithBuckets(
+		"block_duration",
+		namespace,
+		"Duration in ns with update to disk",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000_000, 2, 10),
+	).WithLabelValues()
+	blockDurationWait = metrics.NewHistogramWithBuckets(
+		"block_duration_wait",
+		namespace,
+		"Duration in ns for db to be ready to process transactions.",
+		[]string{},
+		prometheus.ExponentialBuckets(100_000, 2, 10),
+	).WithLabelValues()
+	blockDurationTxs = metrics.NewHistogramWithBuckets(
+		"block_duration_txs",
+		namespace,
+		"Duration in ns to parse and execute transactions.",
+		[]string{},
+		prometheus.ExponentialBuckets(1_000_000, 2, 10),
+	).WithLabelValues()
+	blockDurationRewards = metrics.NewHistogramWithBuckets(
+		"block_duration_rewards",
+		namespace,
+		"Duration in ns to compute and update rewards.",
+		[]string{},
+		prometheus.ExponentialBuckets(1_000_000, 2, 10),
+	).WithLabelValues()
+	blockDurationPersist = metrics.NewHistogramWithBuckets(
+		"block_duration_persist",
+		namespace,
+		"Duration in ns to write updates on disk.",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000_000, 2, 10),
+	).WithLabelValues()
+)
+
+var (
+	stagingCacheSize = metrics.NewGauge(
```

Comment:
unused?

Reply:
thank, it was basically the same as one below so i removed it